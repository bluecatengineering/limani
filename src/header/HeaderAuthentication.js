/*
Copyright 2023-2024 BlueCat Networks Inc.

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.
*/
import PropTypes from 'prop-types';
import usePlatformData from '../hooks/usePlatformData';

import './HeaderAuthentication.less';

/**
 * HeaderAuthentication component displays the connected Authentication
 * service alias and URL. <br>
 * This component is intended to be nested inside the PlatformDataContext
 * as it will require access to PlatformData.
 */
const HeaderAuthentication = ({ className }) => {
    const { data } = usePlatformData();
    const authenticationAlias = data?.user?.authentication_info?.alias;
    const authenticationLink = data?.user?.authentication_info?.url;
    const authenticationService = data?.user?.authentication_info?.service;

    return (
        <>
            {authenticationService && (
                <div
                    className={`HeaderAuthentication__authentication${
                        className ? ` ${className}` : ''
                    }`}>
                    <span>
                        {authenticationAlias ? (
                            <a
                                target='_blank'
                                rel='noreferrer'
                                href={authenticationLink}>
                                {authenticationService}:{' '}
                                <u>{authenticationAlias}</u>
                            </a>
                        ) : (
                            authenticationService
                        )}
                    </span>
                </div>
            )}
        </>
    );
};

HeaderAuthentication.propTypes = {
    /** The component class name(s). */
    className: PropTypes.string,
};

export default HeaderAuthentication;
