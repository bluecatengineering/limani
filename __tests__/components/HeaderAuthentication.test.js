/*
Copyright 2023 BlueCat Networks Inc.

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.
*/

import { shallow } from 'enzyme';
import HeaderAuthentication from '../../src/header/HeaderAuthentication';

jest.unmock('../../src/header/HeaderAuthentication');

const mockBAM = {
    data: {
        user: {
            // eslint-disable-next-line camelcase
            authentication_info: {
                alias: 'BAM-9.5.0',
                url: '#',
                service: 'BAM',
            },
        },
    },
};
const mockMicetro = {
    data: {
        user: {
            // eslint-disable-next-line camelcase
            authentication_info: {
                alias: 'Micetro-10',
                url: '#',
                service: 'Micetro',
            },
        },
    },
};
const mockStandalone = {
    data: {
        user: {
            // eslint-disable-next-line camelcase
            authentication_info: null,
        },
    },
};
jest.mock('../../src/hooks/usePlatformData', () => {
    const mockValue = jest.fn();
    const mockUsePlatformData = jest.fn(() => mockValue);
    mockUsePlatformData.mockImplementation(() => mockValue);
    return mockUsePlatformData;
});
const mockUsePlatformData = require('../../src/hooks/usePlatformData');
mockUsePlatformData.mockReturnValueOnce({ ...mockMicetro });
mockUsePlatformData.mockReturnValueOnce({ ...mockStandalone });
mockUsePlatformData.mockReturnValue({ ...mockBAM });

describe('HeaderAuthentication', () => {
    describe('Rendering', () => {
        it('Render HeaderAuthentication component with Micetro authentication', () => {
            const wrapper = shallow(<HeaderAuthentication />);
            expect(wrapper.getElement()).toMatchSnapshot();
        });

        it('Render HeaderAuthentication component with Standalone authentication', () => {
            const wrapper = shallow(<HeaderAuthentication />);
            expect(wrapper.getElement()).toMatchSnapshot();
        });

        it('Render HeaderAuthentication component with BAM authentication and default props', () => {
            const wrapper = shallow(<HeaderAuthentication />);
            expect(wrapper.getElement()).toMatchSnapshot();
        });

        it('Render HeaderAuthentication component with BAM authentication and props', () => {
            const wrapper = shallow(
                <HeaderAuthentication className='varClassName' />,
            );
            expect(wrapper.getElement()).toMatchSnapshot();
        });
    });
});
