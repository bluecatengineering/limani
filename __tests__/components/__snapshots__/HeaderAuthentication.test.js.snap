// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderAuthentication Rendering Render HeaderAuthentication component with BAM authentication and default props 1`] = `
<React.Fragment>
  <div
    className="HeaderAuthentication__authentication"
  >
    <span>
      <a
        href="#"
        rel="noreferrer"
        target="_blank"
        title="BAM-9.5.0"
      >
        BAM
        :
         
        <u>
          BAM-9.5.0
        </u>
      </a>
    </span>
  </div>
</React.Fragment>
`;

exports[`HeaderAuthentication Rendering Render HeaderAuthentication component with BAM authentication and props 1`] = `
<React.Fragment>
  <div
    className="HeaderAuthentication__authentication varClassName"
  >
    <span>
      <a
        href="#"
        rel="noreferrer"
        target="_blank"
        title="BAM-9.5.0"
      >
        BAM
        :
         
        <u>
          BAM-9.5.0
        </u>
      </a>
    </span>
  </div>
</React.Fragment>
`;

exports[`HeaderAuthentication Rendering Render HeaderAuthentication component with Micetro authentication 1`] = `
<React.Fragment>
  <div
    className="HeaderAuthentication__authentication"
  >
    <span>
      <a
        href="#"
        rel="noreferrer"
        target="_blank"
        title="Micetro-10"
      >
        Micetro
        :
         
        <u>
          Micetro-10
        </u>
      </a>
    </span>
  </div>
</React.Fragment>
`;

exports[`HeaderAuthentication Rendering Render HeaderAuthentication component with Standalone authentication 1`] = `
<React.Fragment>
  <div
    className="HeaderAuthentication__authentication"
  >
    <span>
      Standalone
    </span>
  </div>
</React.Fragment>
`;

exports[`HeaderAuthentication Rendering Render HeaderAuthentication component with disable show authentication 1`] = `<React.Fragment />`;
